transport/wsn/src/main/java/org/kaazing/gateway/transport/wsn/WsnAcceptProcessor.java

```diff
@@ -32,9 +34,17 @@ protected Object getMessageFromWriteRequest(WsnSession session, WriteRequest req
         boolean unwrapWsMessages = session.getLocalAddress().getOption(LIGHTWEIGHT);
         if (unwrapWsMessages && message instanceof WsMessage) {
             WsMessage wsMessage = (WsMessage)message;
-            final IoBufferEx ioBufferEx = ((WsMessage) message).getBytes();
+            IoBufferEx ioBufferEx = ((WsMessage) message).getBytes();
             if (wsMessage.getKind() == WsMessage.Kind.TEXT && ioBufferEx instanceof WsBuffer) {
                 ((WsBuffer) ioBufferEx).setKind(WsBuffer.Kind.TEXT);
+            } else if (wsMessage.getKind() == WsMessage.Kind.PING) {
+                WsBuffer wsBuffer = session.getBufferAllocator().wrap(ioBufferEx.buf());
+                wsBuffer.setKind(WsBuffer.Kind.PING);
+                ioBufferEx = wsBuffer;
+            } else if (wsMessage.getKind() == WsMessage.Kind.PONG) {
+                WsBuffer buf = session.getBufferAllocator().wrap(ioBufferEx.buf());
+                buf.setKind(WsBuffer.Kind.PONG);
+                ioBufferEx = buf;
             }
```

Comment:
I think we should have both cases `WsBuffer` or just `IoBufferEx` for all three cases, `TEXT`, `PING` and `PONG`.